_posts/2020-12-10-webpack-vs-sprockets.markdown

```diff
@@ -0,0 +1,313 @@
+
+```js
+// config/webpacker.yml
+
+production:
+  extract_css: true
+```
+
+You can have that as `false` during development or testing, but it's needed for production. I'd recommend you at least try with this as `true` during development to verify it will create the right files for production.
+
+### Images
+
+If you open your `application.js` file, you will find this comment:
+
+```js
+// packs/application.js
+
+// Uncomment to copy all static images under ../images to the output folder and reference
+// them with the image_pack_tag helper in views (e.g <%= image_pack_tag 'rails.png' %>)
+// or the `imagePath` JavaScript helper below.
+//
+// const images = require.context('../images', true)
+// const imagePath = (name) => images(name, true)
+```
+
+You can tell Webpack to compile and emit all the files at `../images` un-commenting the first line:
+
+```js
+// packs/application.js
+
+require.context('../images', true)
+```
+
+Now you can put all your images in `app/javascript/images` and Webpack will move them to `public/packs/media/images` during compilation. You can check at `config/webpacker.yml` which `static_assets_extensions` it supports (many image and font types by default). You can add more extensions there to support more file types.
+
+You can reference assets from your `erb` templates using the `asset_pack_path/url` and `image_pack_path/url` helper methods provided by Webpacker.
+
+> Note that for this configuration to work you must NOT have an `application.css` pack! There is an [open issue](https://github.com/rails/webpacker/issues/2144#issuecomment-745745635) in Webpacker's repo.
+
+### Wrapping up
+
+A simple basic configuration to handle JS, CSS, Images and Fonts using webpacker could be:
+
+```
+app
+  |_ javascript
+    |_ packs
+      |_ application.js
+    |_ src
+      |_ some_javascript.js
+    |_ stylesheets
+      |_ application.css
+    |_ images
+      |_ logo.jpg
+      |_ test.png
+    |_ fonts
+      |_ myfont.ttf
+```
+
+And your `application.js` would look like this:
+
+```js
+// packs/application.js
+
+import '../stylesheets/application.css'
+require.context('../images', true) 
+require.context('../fonts', true)
+
+// and now your JavaScript ...
+import SomeModule from '../src/some_javascript'
+```
+
+You can rename the `source_path` to `assets` and use `javascript` instead of `src` and it will look pretty similar to the Assets Pipeline!
+
+## Pros And Cons
+
+So, what should you use? Sprockets? Webpack? Both? As always, the answer is... it depends. By default, since Rails 6, it uses both: Webpack for JavaScript, Sprocket for any other asset. According to your needs, you may want to use one or the other or change which assets you handle with each of them. You can even handle assets of the same type using both solutions at the same time!
```

Comment:
```suggestion
So, what should you use? Sprockets? Webpack? Both? As always, the answer is... it depends. By default, since version 6, Rails uses both: Webpack for JavaScript, Sprocket for any other asset. According to your needs, you may want to use one or the other or change which assets you handle with each of them. You can even handle assets of the same type using both solutions at the same time!
```